SteamKit2/SteamKit2/Steam/Handlers/SteamUnifiedMessages/SteamUnifiedMessages.cs

```diff
@@ -103,12 +105,27 @@ public override void HandleMsg( IPacketMsg packetMsg )
                 return;
 
             var serviceName = jobName[ ..dot ].ToString();
-            if (!_handlers.TryGetValue( serviceName, out var handler ) )
-                return;
-
-            var methodName = jobName[ ( dot + 1 )..hash ].ToString();
 
-            handler.HandleMsg( methodName, packetMsg );
+            switch ( packetMsgProto.MsgType )
+            {
+                case EMsg.ServiceMethodResponse:
+                {
+                    if (!_handlers.TryGetValue( serviceName, out var handler ) )
+                        return;
+                    var methodName = jobName[ ( dot + 1 )..hash ].ToString();
+                    handler.HandleMsg( methodName, packetMsg );
+                    break;
+                }
+                case EMsg.ServiceMethod:
+                {
+                    if ( !int.TryParse( jobName[ ( hash + 1 ).. ], NumberFormatInfo.InvariantInfo, out var version ) )
+                        return;
+                    var methodName = jobName[ ( dot + 1 )..hash ].ToString();
+                    var notification = new ServiceMethodNotification( packetMsgProto, serviceName, methodName, version );
```

Comment:
So i'm thinking we can still generate typed messages with HandleMsg, but it would have to construct from the request not the response body (since it's NoResponse).